pkg/distributor/distributor_test.go

```diff
@@ -2512,44 +2572,74 @@ func TestDistributor_MetricsMetadata(t *testing.T) {
 	}
 
 	for testName, testData := range tests {
+		testData := testData
+
 		t.Run(testName, func(t *testing.T) {
-			// Create distributor
-			ds, _, _, _ := prepare(t, prepConfig{
-				numIngesters:     numIngesters,
-				happyIngesters:   numIngesters,
-				numDistributors:  1,
-				shuffleShardSize: testData.shuffleShardSize,
-				limits:           nil,
-			})
+			t.Parallel()
 
-			// Push metadata
-			ctx := user.InjectOrgID(context.Background(), "test")
+			for _, ingestStorageEnabled := range []bool{false, true} {
+				ingestStorageEnabled := ingestStorageEnabled
 
-			req := makeWriteRequest(0, 0, 10, false, true, "foo")
-			_, err := ds[0].Push(ctx, req)
-			require.NoError(t, err)
+				t.Run(fmt.Sprintf("ingest storage enabled: %t", ingestStorageEnabled), func(t *testing.T) {
+					t.Parallel()
 
-			// Check how many ingesters are queried as part of the shuffle sharding subring.
-			//nolint:staticcheck
-			replicationSet, err := ds[0].getIngesterReplicationSetForQuery(ctx)
-			require.NoError(t, err)
-			assert.Equal(t, testData.expectedIngesters, len(replicationSet.Instances))
+					testConfig := prepConfig{
+						numIngesters:    numIngesters,
+						happyIngesters:  numIngesters,
+						numDistributors: 1,
+					}
 
-			// Assert on metric metadata
-			metadata, err := ds[0].MetricsMetadata(ctx, client.DefaultMetricsMetadataRequest())
-			require.NoError(t, err)
+					if ingestStorageEnabled {
+						testConfig.ingestStorageEnabled = true
+						testConfig.ingestStoragePartitions = numIngesters
+						testConfig.limits = prepareDefaultLimits()
+						testConfig.limits.IngestionPartitionsTenantShardSize = testData.shuffleShardSize
```

Comment:
aren't these being done in `prepare` already?

https://github.com/grafana/mimir/blob/ac9f6d431cc306b2e557429a10f6fbd9ac11b5f5/pkg/distributor/distributor_test.go#L4002-L4005

you can simplify this if/else by a lot

Reply:
Here I'm setting `IngestionPartitionsTenantShardSize`, and not `IngestionTenantShardSize`. When I test the ingest storage I (intentionally) don't want to set `IngestionTenantShardSize`, and viceversa when testing with the ingest storage disabled, in order to make sure the right config option is used for shuffle sharding. For this reason, I would need the `if` condition even if I will move `IngestionPartitionsTenantShardSize` as a `prepConfig` (due to the lack of a ternary operator in go).

_I'm going to keep it as is for now._

Reply:
yeah makes sense. I didn't read the full config option